prow/bugzilla/client.go

```diff
@@ -205,6 +257,9 @@ func cloneBugStruct(bug *Bug, comments []Comment) *BugCreate {
 		TargetMilestone: bug.TargetMilestone,
 		Version:         bug.Version,
 	}
+	if len(subcomponents) > 0 {
```

Comment:
I don't think the check is needed, you can just undonditionally set the subcomponents as you construct a new `BugCreate` struct anyways

Reply:
I think this might lead to errors when creating a bug on a server that does not have the `sub_components` type. However, a `nil` check might be better.

Reply:
if the thing is nil you just put a nilpointer into the field which won't get serialized. I guess the current approach makes sense to cover cases where we have an empty but non-nil `subcomponents` map as that would get serialized?

Reply:
I made the `GetSubcomponentsOnBug` function return an empty map for both nil and empty map responses from the bugzilla server. The current approach makes sense to ignore both empty and non-existent `sub_components` now.

Reply:
Yeah I agree the current approach makes sense to deal with empty objects but why does that require `GetSubcomponentsOnBug` to return an empty map rather than nil?